### PR TITLE
Update README.md for V4

### LINKED ISSUE #1451: please update README.md to v4

### PR DESCRIPTION
**Pull request by sivapalan**
Closes #1449 
Closes #1451

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,14 +12,15 @@ When Git 2.18 or higher is not in your PATH, falls back to the REST API to downl
...
+- Updated default runtime to node20
+  - This requires a minimum Actions Runner version of [v2.308.0](https://github.com/actions/runner/releases/tag/v2.308.0).
+- Added support for fetching without the `--progress` option
 
 # Usage
 
 <!-- start usage -->
 ```yaml
-- uses: actions/checkout@v3
+- uses: actions/checkout@v4
```
Comment (rasa) [inline README.md:23]: Shouldn't this include a mention of the new option:
`progress: true`

Comment (sivapalan) [reply README.md:23]: It's already been added:
https://github.com/actions/checkout/blob/7f2542429bbbdf913136f117c2110e942d769b0d/README.md?plain=1#L95-L97

**Review by Spiritaine — COMMENTED** [review]
Thanks

**Review by linkicloud — APPROVED** [review]
@sivapalan

**Pull request merged**